pwiz_tools/Skyline/Executables/SkylineBatch/SkylineBatch/RemoteFileControl.cs

```diff
@@ -13,28 +20,80 @@ public partial class RemoteFileControl : UserControl
         private readonly string _downloadFolder;
         private readonly bool _fileRequired;
         private readonly bool _preferPanoramaSource;
+        private readonly bool _templateFile;
 
-        public RemoteFileControl(IMainUiControl mainControl, SkylineBatchConfigManagerState state, Server editingServer, string downloadFolder, bool fileRequired, bool preferPanoramaSource)
+        public RemoteFileControl(IMainUiControl mainControl, SkylineBatchConfigManagerState state, Server editingServer, string downloadFolder, bool fileRequired, bool preferPanoramaSource, bool templateFile = false)
         {
             InitializeComponent();
+            Bitmap bmp = (Bitmap)this.btnOpenFromPanorama.Image;
+            bmp.MakeTransparent(bmp.GetPixel(0, 0));
 
             State = state;
             _mainControl = mainControl;
             _lastStelectedIndex = -1;
             _downloadFolder = downloadFolder;
             _fileRequired = fileRequired;
             _preferPanoramaSource = preferPanoramaSource;
+            _templateFile = templateFile;  
             UpdateRemoteSourceList();
-
             if (editingServer != null)
             {
                 textRelativePath.Text = editingServer.RelativePath;
                 comboRemoteFileSource.SelectedItem = editingServer.FileSource.Name;
             }
+            CheckIfPanoramaSource();
+
         }
 
         public SkylineBatchConfigManagerState State { get; private set; }
 
+        private RemoteFileSource GetRemoteFileSource(){
+
+            RemoteFileSource remoteFileSource;
+            try
+            {
+                remoteFileSource = RemoteFileSourceFromUi();
+            }
+            catch
+            {
+                return null;
+            }
+
+            return remoteFileSource;
+
+        }
+
+        private void CheckIfPanoramaSource()
+        {
+            RemoteFileSource source = GetRemoteFileSource();
+            if (source == null)
+            {
+                ShowPanoramaBtn(false);
+                return;
+            }
+
+            Uri uri = source.URI;
+            UserState state = PanoramaUtil.ValidateServerAndUser(ref uri,source.Username,source.Password);
+            ShowPanoramaBtn(state.IsValid());
+
+        }
+
+        private void ShowPanoramaBtn(bool show)
+        {
+            if (show)

```

Comment:
You might want to add:
```
if (show == btnOpenFromPanorama.Visible)
    return;
```
so that if someone calls "ShowPanoramaBtn" when the thing is already in the correct state the width of textRelativePath does not get changed.

Reply:
Thanks for the feedback, I made a change implementing this.